symfony/plugins/orangehrmAdminPlugin/modules/admin/actions/deletePayGradesAction.class.php

```diff
@@ -45,7 +45,9 @@ public function execute($request) {
                     foreach ($toBeDeletedPayGradeIds as $toBeDeletedPayGradeId) {
 
                         $payGrade = $this->getPayGradeService()->getPayGradeById($toBeDeletedPayGradeId);
-                        $payGrade->delete();
+                        if ($payGrade instanceof PayGrade) {
+                            $payGrade->delete();
+                        }
                     }
                     $this->getUser()->setFlash('success', __(TopLevelMessages::DELETE_SUCCESS));
```

Comment:
flash looks like it will be set 20 times if 20 Ids are passed?

Reply:
But the flash message set outside of the for loop.